native-maven-plugin/src/functionalTest/groovy/org/graalvm/buildtools/maven/MavenTestExecutionFunctionalTests.groovy

```diff
@@ -0,0 +1,71 @@
+ *
+ * The Universal Permissive License (UPL), Version 1.0
+ *
+ * Subject to the condition set forth below, permission is hereby granted to any
+ * person obtaining a copy of this software, associated documentation and/or
+ * data (collectively the "Software"), free of charge and under any and all
+ * copyright rights in the Software, and any and all patent rights owned or
+ * freely licensable by each licensor hereunder covering either (i) the
+ * unmodified Software as contributed to or provided by such licensor, or (ii)
+ * the Larger Works (as defined below), to deal in both
+ *
+ * (a) the Software, and
+ *
+ * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
+ * one is included with the Software each a "Larger Work" to which the Software
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ *
+ * The above copyright notice and either this complete permission notice or at a
+ * minimum a reference to the UPL must be included in all copies or substantial
+ * portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.graalvm.buildtools.maven
+
+class MavenTestExecutionFunctionalTests extends AbstractGraalVMMavenFunctionalTest {
+    def "test if the skipTestExecution flag skips execution of tests"() {
+        withSample("junit-tests")
+
+        when:
+        mvn '-DquickBuild', '-DskipTestExecution=true', '-Pnative', 'test'
+
+        then:
+        buildSucceeded
+        outputContains "GraalVM Native Image: Generating 'native-tests' (executable)"
+        outputDoesNotContain "containers found"
+        outputDoesNotContain "containers skipped"
+        outputDoesNotContain "containers started"
+        outputDoesNotContain "tests found"
+        outputDoesNotContain "tests skipped"
+        outputDoesNotContain "tests started"
+    }
+
+    def "test if setting the failNoTests flag allows build to continue if no tests are present"() {
+        withSample("non-native")
+
+        when:
+        mvn '-DquickBuild', '-DfailNoTests=false', '-Pnative', 'test'
+
+        then:
+        buildSucceeded
```

Comment:
Maybe add expected output that reflects your changes (from [here](https://github.com/graalvm/native-build-tools/pull/720/files#diff-407c562b048d82141f278e817ad7f1603db1be9c062a6e08816925d4358c0a89R175-R181))

Reply:
good callout! Done.